homeassistant/components/renault/renault_vehicle.py

```diff
@@ -0,0 +1,126 @@
+            "model": (details.get_model_label() or "").capitalize(),
+            "name": details.registrationNumber or "",
+            "sw_version": details.get_model_code() or "",
+        }
+        self.coordinators: dict[str, RenaultDataUpdateCoordinator] = {}
+        self.hvac_target_temperature = 21
+        self._scan_interval = scan_interval
+
+    @property
+    def details(self) -> models.KamereonVehicleDetails:
+        """Return the specs of the vehicle."""
+        return self._details
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        return self._device_info
+
+    async def async_initialise(self) -> None:
+        """Load available sensors."""
+        if await self.endpoint_available("cockpit"):
+            self.coordinators["cockpit"] = RenaultDataUpdateCoordinator(
+                self.hass,
+                LOGGER,
+                # Name of the data. For logging purposes.
+                name=f"{self.details.vin} cockpit",
+                update_method=self.get_cockpit,
+                # Polling interval. Will only be polled if there are subscribers.
+                update_interval=self._scan_interval,
+            )
+        if await self.endpoint_available("hvac-status"):
+            self.coordinators["hvac_status"] = RenaultDataUpdateCoordinator(
+                self.hass,
+                LOGGER,
+                # Name of the data. For logging purposes.
+                name=f"{self.details.vin} hvac_status",
+                update_method=self.get_hvac_status,
+                # Polling interval. Will only be polled if there are subscribers.
+                update_interval=self._scan_interval,
+            )
+        if self.details.uses_electricity():
+            if await self.endpoint_available("battery-status"):
+                self.coordinators["battery"] = RenaultDataUpdateCoordinator(
+                    self.hass,
+                    LOGGER,
+                    # Name of the data. For logging purposes.
+                    name=f"{self.details.vin} battery",
+                    update_method=self.get_battery_status,
+                    # Polling interval. Will only be polled if there are subscribers.
+                    update_interval=self._scan_interval,
+                )
+            if await self.endpoint_available("charge-mode"):
+                self.coordinators["charge_mode"] = RenaultDataUpdateCoordinator(
+                    self.hass,
+                    LOGGER,
+                    # Name of the data. For logging purposes.
+                    name=f"{self.details.vin} charge_mode",
+                    update_method=self.get_charge_mode,
+                    # Polling interval. Will only be polled if there are subscribers.
+                    update_interval=self._scan_interval,
+                )
+        for key in list(self.coordinators.keys()):
```

Comment:
```suggestion
        for key in self.coordinators:
```

Reply:
`list(xxx)` is still required to avoid `RuntimeError: dictionary changed size during iteration`